src/DynamoCore/Nodes/CodeBlockNode.cs

```diff
@@ -310,14 +311,15 @@ internal override IEnumerable<AssociativeNode> BuildAst(List<AssociativeNode> in
 
             foreach (var astNode in codeStatements.Select(stmnt => NodeUtils.Clone(stmnt.AstNode)))
             {
-                MapIdentifiers(astNode);
+                if (context != AstBuilder.CompilationContext.NodeToCode)
+                    MapIdentifiers(astNode);
                 resultNodes.Add(astNode as AssociativeNode);
             }
 
             return resultNodes;
         }
 
-        public override IdentifierNode GetAstIdentifierForOutputIndex(int portIndex)
+        private IdentifierNode GetAstIdentifierForOutputIndexInternal(int portIndex, bool forRawName)
```

Comment:
Another nasty part. For code block node, we map all variables to an unique name. `GetAstIdentifierForOutputIndex()` returns mapped variable name for code block node, which is not what I want here -- I need its original name like `x` so that if there are two code block nodes both have `x`, then one of `x` should be renamed. Otherwise mapped variable is always unique. But yes, I will add summary for this function.